app/routes.rb

```diff
@@ -120,6 +124,10 @@ def repo
   Bibliotheca::Collection::Guides.find(params['id']).as_json
 end
 
+get '/guides/:gid/exercises/:eid/test' do
+  Bibliotheca::Collection::Guides.find(params['gid']).run_tests(params['eid'].to_i).as_json
```

Comment:
I prefer to use `guide_id` and `exercise_id` instead of `eid` or things like that. It is more intention revealing.